### DISCUSSION TITLE
does the DecisionTreeEncoder work univariate

### DISCUSSION THREAD

**Discussion by csetzkorn** [feature-engine/feature_engine#409] [Q&A] [answered]
Curious, does the DecisionTreeEncoder (see [here](https://feature-engine.readthedocs.io/en/1.1.x/encoding/DecisionTreeEncoder.html)) work univariate-ly as the doc seems to suggest (i.e. for every feature separately) or does it consider interactions as it should be? I hope this makes sense? Thanks.

**Comment by solegalli** [comment, marked as answer]
It works univariately.

**Comment by csetzkorn** [comment]
Thanks for the reply. Will accept as answer. So this is with the expectation for the upstream model to squeeze more signal out of this? IMHO information could be lost though? I am more an R person and have used decision trees in the past and simply fed in the features and used the terminal numbers, which are ordinal w.r.t. the target for regression. In Python, it appears that DecisionTreeRegressor cannot deal with categorical/object types and e.g. one hot has to be used, which is a bit silly IMHO when one deals with very granular categorical features. Are you aware of decision trees (for regression) that can deal with categorical/object types in Python? Thanks.

**Reply by solegalli** [reply]
A model that can handle categorical variables out of the box is catboost. And it has an implementation in Python.

Handling categorical features manually vs letting the model do it has its trade-offs. If you do it manually, you know exactly what you are doing and you can actually modify the behaviour to something that suits your problem. If the model do it for you, I'd recommend knowing what is going on with the categories under the hood.

**Reply by csetzkorn** [reply]
I guess what I am say-ing is that doing it univariate loses information such as interaction effects ...
